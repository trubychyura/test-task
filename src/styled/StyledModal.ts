import styled from 'styled-components';
import { Col, Image, Modal, Row } from 'react-bootstrap';

export const StyledModal = styled(Modal)`
  @media (max-width: ${({ theme }) => theme.media.md}) {
    padding: 0 !important;
  }

  .modal-title {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }

  .modal-dialog {
    max-width: 60%;
    @media (max-width: ${({ theme }) => theme.media.lg}) {
      max-width: 70%;
    }
    @media (max-width: ${({ theme }) => theme.media.md}) {
      max-width: 90%;
      margin: auto;
    }
  }
`;

export const StyledImage = styled(Image)`
  display: block;
  max-height: 300px;
  max-width: 100%;
  margin: auto;
`;

export const StyledDescription = styled(Col)`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const StyledComments = styled(Col)`
  padding: 15px;
`;

export const StyledComentHeader = styled(Row)`
  margin: 0;
  justify-content: space-between;
`;
