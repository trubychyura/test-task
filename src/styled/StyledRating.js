import styled from 'styled-components';
import { Star } from '@styled-icons/boxicons-solid/Star';
import { Form } from 'react-bootstrap';

export const StyledStar = styled(Star)`
  cursor: pointer;
  transition: color 200ms;
  width: 25px;
  heidth: 25px;

  ${({ value, rating, hover }) =>
    value <= (hover || rating) ? `color: gold;` : `color: silver;`}

  @media (max-width: 580px) {
    width: 18px;
    heidht: 18px;
  }
`;

export const StyledRadio = styled(Form.Check)`
  display: none;
`;

export const StyledContainer = styled('div')`
  display: flex;
`;
