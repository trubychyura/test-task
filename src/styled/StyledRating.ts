import styled from 'styled-components';
import { Star } from '@styled-icons/boxicons-solid/Star';
import { Form } from 'react-bootstrap';
import { StarProps } from '../types';

export const StyledStar = styled(Star)<StarProps>`
  transition: color 200ms;
  width: 25px;
  heidth: 25px;
  ${({ value, rating, hover, theme }) =>
    value <= (hover || rating)
      ? `color: ${theme.color.gold};`
      : `color: ${theme.color.silver};`}
  ${({ hover }) => hover !== undefined && `cursor: pointer;`}

  @media (max-width: ${({ theme }) => theme.media.md}) {
    width: 18px;
    heidht: 18px;
  }
`;

export const StyledRadio = styled(Form.Check)`
  display: none;
`;

export const StyledContainer = styled('div')`
  display: flex;
`;
