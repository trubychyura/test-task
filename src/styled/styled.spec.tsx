import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';

import theme from './theme';
import { StyledModal } from './StyledModal';
import { StyledStar } from './StyledRating';

type setUpProps = {
  value?: number;
  rating?: number;
  hover?: number;
};
describe('Styled components', () => {
  let component;
  const setUp = (Comp: any, props?: setUpProps): ShallowWrapper =>
    shallow(<Comp {...props} theme={theme} />);

  describe('Styled Modal', () => {
    it('should render Styled Modal', () => {
      expect(setUp(StyledModal)).toMatchSnapshot();
    });
  });

  describe('Styled Star', () => {
    it('should render Styled Star', () => {
      expect(setUp(StyledStar)).toMatchSnapshot();
    });

    it('should render Star with silver color', () => {
      component = setUp(StyledStar, { value: 5, rating: 2 });
      expect(component).toHaveStyleRule('color', theme.color.silver);
    });

    it('should render Star with gold color', () => {
      component = setUp(StyledStar, { value: 2, hover: 5 });
      expect(component).toHaveStyleRule('color', theme.color.gold);
    });
  });
});
