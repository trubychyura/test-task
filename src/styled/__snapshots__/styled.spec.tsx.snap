// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled components Styled Modal should render Styled Modal 1`] = `
.c0 .modal-title {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c0 .modal-dialog {
  max-width: 60%;
}

@media (max-width:580px) {
  .c0 {
    padding: 0 !important;
  }
}

@media (max-width:1000px) {
  .c0 .modal-dialog {
    max-width: 70%;
  }
}

@media (max-width:580px) {
  .c0 .modal-dialog {
    max-width: 90%;
    margin: auto;
  }
}

<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  className="c0"
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  restoreFocus={true}
  show={false}
  theme={
    Object {
      "color": Object {
        "gold": "gold",
        "silver": "silver",
      },
      "media": Object {
        "lg": "1000px",
        "md": "580px",
      },
    }
  }
/>
`;

exports[`Styled components Styled Star should render Styled Star 1`] = `
.c0 {
  -webkit-transition: color 200ms;
  transition: color 200ms;
  width: 25px;
  heidth: 25px;
  color: silver;
}

@media (max-width:580px) {
  .c0 {
    width: 18px;
    heidht: 18px;
  }
}

<Star
  className="c0"
  theme={
    Object {
      "color": Object {
        "gold": "gold",
        "silver": "silver",
      },
      "media": Object {
        "lg": "1000px",
        "md": "580px",
      },
    }
  }
/>
`;
